core/src/main/java/org/streamprocessor/core/transforms/TransformMessageFn.java

```diff
@@ -58,21 +71,48 @@ public void processElement(@Element PubsubMessage received, OutputReceiver<Pubsu
                             .getJSONObject("source")
                             .getString("provider");
 
+            LocalDate currentDate = LocalDate.now(ZoneId.of("Europe/Stockholm"));
+
+            if (dataContract.isNull("valid_from")) {
+                throw new CustomExceptionsUtils.MissingMetadataException(
+                        "No `valid_from` found in data contract");
+            } else {
+                String validFrom = dataContract.getString("valid_from");
+                LocalDate validFromDate = LocalDate.parse(validFrom);
+
+                if (currentDate.isBefore(validFromDate)) {
+                    throw new CustomExceptionsUtils.InactiveDataContractException(
+                            "Data contract is not valid for the current time. "
+                                    + "Data contract is valid from: "
+                                    + validFrom);
+
+                } else if (dataContract.isNull("valid_to")) {
+                    String validTo = dataContract.getString("valid_to");
```

Comment:
if `valid_to` is null then we shouldn't fetch it from the contract right?

Shouldn't it be something like:

```java
else if (!dataContract.isNull("valid_to")) {
  String validTo = dataContract.getString("valid_to");
  ...
}
```

Reply:
yes, thanks!